### DISCUSSION TITLE
Failed to JanusGraphFactory.drop(graph);

### DISCUSSION THREAD

**Discussion by zingdle** [JanusGraph/janusgraph#2924] [Q&A] [answered]
Hi all, I'm new to janusgraph. When I'm droping the database, the cassandra log  `drop keyspace janusgraph` and then the driver throw exceptions. Then a few seconds later,  the janusgraph complains about `keyspace janusgraph not exists` repeatly. 

Is this related to https://github.com/JanusGraph/janusgraph/issues/1313?  Thanks in advance!

---

- Version:  6.0
- Storage Backend:  cql
- Mixed Index Backend:  elasticsearch
- Steps to Reproduce: 

set up docker environment:

```
# go to https://github.com/JanusGraph/janusgraph-docker
docker-compose -f docker-compose-cql-es.yml up
```

driver code:

```java
JanusGraph graph = JanusGraphFactory.build().
        set("storage.backend", "cql").
        set("storage.hostname", "MY_IP").
        set("storage.cql.keyspace", "janusgraph").
        set("storage.cql.local-datacenter", "datacenter1").
        set("cache.db-cache", "true").
        set("cache.db-cache-clean-wait", "20").
        set("cache.db-cache-time", "180000").
        set("cache.db-cache-size", "0.25").
        set("index.search.backend", "elasticsearch").
        set("index.search.hostname", "MY_IP").
        open();
JanusGraphFactory.drop(graph);
```

driver error:

```
com.datastax.oss.driver.api.core.DriverTimeoutException: Query timed out after PT2S
	at com.datastax.oss.driver.api.core.DriverTimeoutException.copy(DriverTimeoutException.java:34)
	at com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures.getUninterruptibly(CompletableFutures.java:149)
	at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:53)
	at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:30)
	at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:230)
	at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:54)
	at org.janusgraph.diskstorage.cql.CQLStoreManager.clearStorage(CQLStoreManager.java:278)
	at org.janusgraph.diskstorage.Backend.clearStorage(Backend.java:636)
	at org.janusgraph.core.JanusGraphFactory.drop(JanusGraphFactory.java:229)
```

docker log:

![image](https://user-images.githubusercontent.com/31171119/148508796-ccf30f82-0b5b-4847-be73-57890881b324.png)

**Comment by li-boxuan** [comment]
This is fixed in https://github.com/JanusGraph/janusgraph/issues/2811 which will be released in 0.6.1. As a temporary fix, can you change `storage.cql.request-timeout` to 12 seconds?

**Reply by zingdle** [reply]
Thanks to your quick response! I tried to set `storage.cql.request-timeout` to `12000` or even `120000`, but the error still exists. 

I can confirm that the cassandra is set up correctly becasue `GraphOfTheGodsFactory.load(graph);` works as expected.

**Reply by li-boxuan** [reply]
That is weird; can you try JanusGraph 0.5.3 and see if you encounter the same problem?

**Reply by zingdle** [reply]
Yes, the problem still exists... I'm wondering if the docker environment has something wrong here. Do you have any ideas about what could cause this?

**Comment by zingdle** [comment, marked as answer]
I see what goes wrong here. The demo I wrote sets up ganusgraph connection, but the Dockerfile provided [here](https://github.com/JanusGraph/janusgraph-docker) also sets up janusgraph server in the background and may interfer `drop keyspace` action in cassandra. 

All I have to do is to stop the janusgraph container.
